### PR TITLE
Conditional requests

### PR DESCRIPTION
**Pull request by JacopoDaeli**
Follow #266.

### CODE REVIEW

File: src/octonode/cmd.coffee [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#
+# cmd.coffee: Generic Github command class
+#
+# Copyright © 2011 Pavan Kumar Sunkara. All rights reserved
+#
+
+# Requiring modules
+
+# Initiate class
+class Cmd
```
Comment (pksunkara) [inline src/octonode/cmd.coffee]: Rename file and this to `Base`

File: src/octonode/client.coffee [hunk 2]
```diff
@@ -41,6 +41,11 @@ class Client
     if @token and typeof @token == 'string'
       @requestDefaults.headers.Authorization = "token " + @token
 
+  # Add http conditional "etag"
+  conditional: (etag) ->
```
Comment (pksunkara) [inline src/octonode/client.coffee]: Can you also add documentation on how to use this?

File: src/octonode/cmd.coffee (hunk 1 above)
Comment (JacopoDaeli) [reply src/octonode/cmd.coffee]: 🌮

File: src/octonode/client.coffee (hunk 2 above)
Comment (JacopoDaeli) [reply src/octonode/client.coffee]: 🌮

**Comment by JacopoDaeli** [comment]
@pksunkara should we bump the npm version as well?

**Comment by pksunkara** [comment]
Sure, go ahead with that.

**Comment by JacopoDaeli** [comment]
Should go for `0.9.0` right? Or should we consider this as patch and go for `0.8.1`?

**Comment by pksunkara** [comment]
Minor version is better.

**Comment by JacopoDaeli** [comment]
👍

**Comment by JacopoDaeli** [comment]
@pksunkara changes done. What do you think?

**Comment by pksunkara** [comment]
Hey, Sorry about the misunderstanding. I meant 0.8.1. Not 0.9.0. Could you correct it and squash the last commit? Thanks. Will merge and publish immediately once thats done.

**Comment by JacopoDaeli** [comment]
Sure, will do.

**Comment by JacopoDaeli** [comment]
@pksunkara I can't merge since I do not have `write access` to the repository. Can you merge it?

image

**Comment by JacopoDaeli** [comment]
@pksunkara I updated the `package.json` file 👍 . You can merge!

**Comment by pksunkara** [comment]
> squash the last commit

Please squash the last 3 commits into a single one.

**Comment by JacopoDaeli** [comment]
closing this. opening a new one.

**Comment by JacopoDaeli** [comment]
@pksunkara https://github.com/pksunkara/octonode/pull/292
This should be ok!

**Pull request closed without merging**